pds_doi_core/db_util/doi_database.py

```diff
@@ -0,0 +1,422 @@
+        o_query_string += ',release_date INT '  # as Unix Time, the number of seconds since 1970-01-01 00:00:00 UTC.
+        o_query_string += ',transaction_key TEXT PRIMARY KEY NOT NULL' # transaction (key is node id /datetime) 
+        o_query_string += ',is_latest BOOLEAN NULL); ' # when the transaction is the latest 
+        logger.debug(f"o_query_string {o_query_string}")
+
+        return o_query_string
+
+    def doi_create_q_string_for_insert(self,table_name):
+        ''' Build the query string to insert into the table in the SQLite database. '''
+
+        # Note that this table structure is defined here so if you need to know the structure.
+        o_query_string = 'INSERT INTO ' + table_name + ' '
+
+        o_query_string += '(status'
+        o_query_string += ',latest_update'
+        o_query_string += ',submitter'
+        o_query_string += ',title'
+        o_query_string += ',type'
+        o_query_string += ',subtype'
+        o_query_string += ',node_id'
+        o_query_string += ',lid'
+        o_query_string += ',vid'
+        o_query_string += ',doi'
+        o_query_string += ',release_date'
+        o_query_string += ',transaction_key'
+        o_query_string += ',is_latest) VALUES '
+        o_query_string += '(?,?,?,?,?,?,?,?,?,?,?,?,?)'
+
+        logger.debug(f"o_query_string {o_query_string}")
+
+        return o_query_string
+
+    def doi_create_q_string_for_transaction_insert(self,table_name):
+        ''' Build the query string to insert a transaction into the table in the SQLite database. '''
+
+        # Note that this table structure is defined here so if you need to know the structure.
+        # Also note that we are not setting all columns, just the ones related to a transaction.
+        o_query_string = 'INSERT INTO ' + table_name + ' '
+
+        o_query_string += '(status'
+        o_query_string += ',submitter'
+        o_query_string += ',latest_update'
+        o_query_string += ',node_id'
+        o_query_string += ',transaction_key) VALUES '
+        o_query_string += '(?,?,?,?,?)'
+
+        logger.debug(f"o_query_string {o_query_string}")
+
+        return o_query_string
+
+    def doi_create_table(self,db_name,table_name,drop_exist_table_flag=False):
+        ''' Create a given table in the SQLite database. '''
+
+        logger.debug(f"self.m_my_conn {self.m_my_conn}")
+        if self.m_my_conn is None:
+            logger.warn(f"Connection is None in database {self.get_database_name()}")
+            self.m_my_conn = self.doi_create_connection(db_file)
+
+        o_table_exist_flag = self.check_if_table_exist(table_name)
+        logger.debug(f"o_table_exist_flag {o_table_exist_flag}")
+
+        if o_table_exist_flag:
+            logger.warn(f"Table {table_name} already exist")
+            if (drop_exist_table_flag):
+                logger.debug(f"drop_exist_table_flag is True for table {table_name}")
+                self.doi_drop_table(db_name,table_name)
```

Comment:
I am not sure if we want to do that as that will delete our history if the method is called on an existing database, or you will need to check if the table exist before calling the function (which is already done in this method).

Reply:
Any possibility of removing the table has been removed since it is a rather dangerous action.  Should leave this action to the developer or the administrator.

Reply:
Perfect. Thanks